pyproject.toml

```diff
@@ -1,12 +1,68 @@
 [build-system]
-requires = ['setuptools>=40.8.0']
-build-backend = 'setuptools.build_meta'
+requires = ["setuptools>=61.0.0"]
```

Comment:
So it turns out that [v69.3.0](https://setuptools.pypa.io/en/latest/history.html#v69-3-0) supports PEP 625 and so canonicalizes package names. I think they really mean "normalizes". What this means is that we now get the following files:

```
Django-X.Y-py3-none-any.whl
django-X.Y.tar.gz
```

Note that the sdist filename is now lower-cased. It seems that this is going to be the way forward and that the wheel also ought to be the same, but that remains to be seen.

Reply:
This! I got this during the Django 5.1 alpha release and it was a nightmare (and I was not using this PR).
I started the alpha release on a system that only had Python 3.12, and since `distutils` is gone in 3.12, I had to manually install `setuptools` which brought the lastest version and that resulted in lowercase-named `django` tarball.

Because the lowercase-name `django` generated complains after the release that Sarah did at the beginning of May, and also the Jenkins CI job fails for lowercase `django`, I resorted to using a Python 3.11 docker image to build the release files, but we'd need a plan for this.

Reply:
> Do you suggest we require setuptools>=69.3.0?

I guess we could do that to ensure we always get the normalised name as that looks to be the way it'll be going forward and we won't get a day in it.

Alternatively, we can pin to `setuptools>=61.0.0,<69.3.0` here for now to give time to communicate the change if necessary as, with this PR, we get build isolation. But we'll need to unpin at some point.

> …and also the Jenkins CI job fails for lowercase `django`…

@nessita This sounds suspect. I don't understand - which job is this and why should it care about the name of the sdist? Are we not always installing a wheel via `pip`? That wouldn't care as it will normalise the name anyway.

Reply:
@claudep @ngnpope The issue with the "lowercased-name" django tarball I mentioned above was related to the *confirm release* CI job (sorry I wasn't explicit yesterday, it was a tough day).

At the beginning of May, when @sarahboyce did her first release, she used Python 3.12 and pip-installed the latest `setuptools`. The resulting django tarball was lowercased and that led to "issues" (there was some complaining in Mastodon and a failed CI run for the "confirm-release" job), so I *thought* that the CI job failed because of the "wrong casing" (see [run 1](https://djangoci.com/job/confirm-release/196/console) and [run 2](https://djangoci.com/job/confirm-release/197/console)). **But**, after a chat with Sarah today, she reminded me that the failures were caused by a manual rename of the tarball from `django-...` to `Django-..` (capitalized) so that was the cause of the CI issue.

Long story short, the CI confirm-release job should be OK with either casing.

Then, and as mentioned yesterday, I was also using python 3.12 and latest setuptools to build the binaries, which resulted in the "django-..." named tarball, so I panicked and used a docker image with Python 3.11 (which includes distutils) to complete the release. Having looked at this again with fresher eyes, I could've used Python 3.12 with `pip install --force "setuptools<67"` and that would have let me produce the usual casing "Django-..." tarball and wheel.

In summary, the lowercase django may be fine but unclear if "the community" may complain for reasons I can't think of today.

Reply:
I think you might encounter some complaining from repackagers, whose scripts might need adjusting for the different casing, but I wouldn’t expect that to be a big issue considering it’s a standard and happening to all packages.